app/src/androidTest/java/org/kiwix/kiwixmobile/splash/KiwixSplashActivityTest.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.splash
+
+import android.Manifest
+import android.content.Context
+import androidx.preference.PreferenceManager
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.matcher.IntentMatchers
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.GrantPermissionRule
+import com.adevinta.android.barista.interaction.BaristaSleepInteractions
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import org.kiwix.kiwixmobile.main.KiwixMainActivity
+import org.kiwix.kiwixmobile.testutils.TestUtils
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class KiwixSplashActivityTest {
+
+  @Rule
+  var readPermissionRule: GrantPermissionRule =
+    GrantPermissionRule.grant(Manifest.permission.READ_EXTERNAL_STORAGE)
+
+  @Rule
+  var writePermissionRule: GrantPermissionRule =
+    GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+  private var context: Context? = null
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    context = InstrumentationRegistry.getInstrumentation().targetContext
+  }
+
+  @Test
+  fun testFirstRun() {
+    shouldShowIntro(true)
+    val activityScenario = ActivityScenario.launch(KiwixMainActivity::class.java)
+    activityScenario.onActivity {
```

Comment:
I think it will be a good idea to make it a global variable and re-use it on other tests.